src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -139,12 +178,203 @@ class IndexLogEntryTest extends SparkFunSuite {
             .Columns(Seq("col1"), Seq("col2", "col3")),
           schema.json,
           200)),
-      Content("rootContentPath", Seq()),
+      Content(Directory("rootContentPath")),
       Source(SparkPlan(expectedSourcePlanProperties)),
       Map())
     expected.state = "ACTIVE"
     expected.timestamp = 1578818514080L
 
+    val actual = JsonUtils.fromJson[IndexLogEntry](jsonString)
+    assert(actual.equals(expected))
+  }
+
+  test("Content Test: files api lists all files from Content object.") {
+    val dir2Path = PathUtils.makeAbsolute(nestedDir.toString)
+    val content = Content.fromDirectory(dir2Path)
+
+    val expected = Seq(f3, f4).map(toPath).toSet
+    val actual = content.files.toSet
+    assert(actual.equals(expected))
+  }
+
+  test("Content Test: files api lists all files recursively from Content object.") {
+    val dirPath = PathUtils.makeAbsolute(testDir.toString)
```

Comment:
can't you just utilize `toPath` you introduced? same in other places.

Reply:
thanks @imback82 , fixed